sympy/functions/elementary/tests/test_trigonometric.py

```diff
@@ -1285,11 +1286,14 @@ def test_leading_terms():
     assert sin(S.Half).as_leading_term(x) == sin(S.Half)
     assert cos(1/x).as_leading_term(x) == AccumBounds(-1, 1)
     assert cos(S.Half).as_leading_term(x) == cos(S.Half)
-
-    for func in [tan, cot]:
-        for a in (1/x, S.Half):
-            eq = func(a)
-            assert eq.as_leading_term(x) == eq
+    assert sec(1/x).as_leading_term(x) == AccumBounds(S.NegativeInfinity, S.Infinity)
+    assert csc(1/x).as_leading_term(x) == AccumBounds(S.NegativeInfinity, S.Infinity)
+    assert tan(1/x).as_leading_term(x) == AccumBounds(S.NegativeInfinity, S.Infinity)
+    assert cot(1/x).as_leading_term(x) == AccumBounds(S.NegativeInfinity, S.Infinity)
+
+    for func in (sin, cos, tan, cot):
+        eq = func(S.Half)
+        assert eq.as_leading_term(x) == eq
```

Comment:
What is the purpose of this test?

Reply:
Ohh it was already there , I wasn't too sure if we would like to keep this test or get rid of it and even improve it somehow, and Sidharth ended up suggesting me to add `sin` and `cos` function too ! Originally we were just dealing with `tan` and `cot`

Reply:
Maybe we could keep it too ! What are your thoughts on the same ?

Reply:
CC @jksuom

Reply:
eq is a constant so it is obviously also the leading term. That is what as_leading_term returns. It is not necessary to test it with different kinds of constants.